### PR TITLE
feat(design-system): option picker input

### PR DESCRIPTION
**Pull request by th0rOdinson**

### CODE REVIEW

File: apps/root/src/pages/aggregator/swap-container/components/step1/index.tsx [hunk 1]
```diff
@@ -100,7 +100,6 @@ const SwapFirstStep = ({
     : (selectedRoute &&
         selectedRoute?.buyToken.address === to?.address &&
         formatUnits(selectedRoute.buyAmount.amount, selectedRoute.buyToken.decimals || 18)) ||
-      '0.0' ||
```
Comment (FiboApe) [inline apps/root/src/pages/aggregator/swap-container/components/step1/index.tsx:103]: we want to show the 0.0 on the inputs right?

File: packages/ui-library/src/components/select/index.tsx [hunk 2]
```diff
@@ -179,11 +176,8 @@ function Select<T extends { key: string | number }, H = object>({
       }}
     >
       {!disabledSearch && (
-        <ListSubheader disableGutters>
+        <ListSubheader disableGutters sx={{ marginBottom: SPACING(2.5) }}>
           <TextField
-            size="small"
-            // Autofocus on textfield
-            autoFocus
```
Comment (FiboApe) [inline packages/ui-library/src/components/select/index.tsx:186]: we definitely want this

File: packages/ui-library/src/components/token-picker/index.tsx [hunk 3]
```diff
@@ -176,8 +176,8 @@ const Row: ItemContent<TokenWithBalance, RowData> = (
     <StyledForegroundPaper key={`${token.chainId}-${token.address}`} onClick={() => onClick(tokenWithBalance)}>
       <ContainerBox flex="1" alignItems="center" gap={3}>
         {token.icon}
-        <ContainerBox flexDirection="column" flex="1" alignItems="flex-start">
-          <Typography variant="bodyBold" color={colors[themeMode].typography.typo2}>
+        <ContainerBox flexDirection="column" flex="1" alignItems="flex-start" gap={0.5}>
+          <Typography variant="bodyBold" color={colors[themeMode].typography.typo2} lineHeight={1}>
```
Comment (FiboApe) [inline packages/ui-library/src/components/token-picker/index.tsx:180]: this has a specific different line height?

File: packages/ui-library/src/components/virtualized-list/index.tsx [hunk 4]
```diff
@@ -1,37 +1,37 @@
 import React, { forwardRef } from 'react';
-import { Components, ItemContent, Virtuoso } from 'react-virtuoso';
+import { ItemContent, Virtuoso } from 'react-virtuoso';
 import styled from 'styled-components';
 
-const StyledList = styled.div`
+const StyledList = styled.div<{ $gap?: number }>`
```
Comment (FiboApe) [inline packages/ui-library/src/components/virtualized-list/index.tsx:5]: This should just be a containerBox

File: packages/ui-library/src/components/virtualized-list/index.tsx [hunk 5]
```diff
@@ -1,37 +1,37 @@
...
+const VirtualizedList = <D, C>({ data, itemContent, fetchMore, context, gap }: VirtualizedListProps<D, C>) => (
   <Virtuoso
     style={{ height: '100%' }}
     data={data}
     itemContent={itemContent}
     endReached={fetchMore}
     context={context}
-    components={VirtuosoComponents}
+    components={{
+      List: forwardRef<HTMLDivElement>(function VirtuosoList(props, ref) {
```
Comment (FiboApe) [inline packages/ui-library/src/components/virtualized-list/index.tsx:31]: If we create it outside it doesnt get created on every re-render and saves up on memory

**Comment by FiboApe** [comment]
Screenshot 2024-09-30 at 1 40 52 pm
Paddings of the contacts modal are wrong

Also the contacts cards are missing their hover styles, same for thje options menu options

**Comment by FiboApe** [comment]
Screenshot 2024-09-30 at 1 43 40 pm
Also some logos are having some weird box-shadow, maybe we should roujnd them

File: packages/ui-library/src/components/select/index.tsx (hunk 2 above)
Comment (th0rOdinson) [reply packages/ui-library/src/components/select/index.tsx:186]: This was suggested by the designer in the notion, I'll undo it anyway!

File: packages/ui-library/src/components/token-picker/index.tsx (hunk 3 above)
Comment (th0rOdinson) [reply packages/ui-library/src/components/token-picker/index.tsx:180]: By default it's line height is 2, so in order to make the line thiner I made that change. With default line height:
Screenshot 2024-09-30 at 1 54 42 PM

**Review by FiboApe — APPROVED** [review]

**Pull request merged**